src/docfx/build/xref/InternalXrefMapBuilder.cs

```diff
@@ -43,13 +43,14 @@ internal class InternalXrefMapBuilder
         _redirectionProvider = redirectionProvider;
     }
 
-    public IReadOnlyDictionary<string, InternalXrefSpec[]> Build()
+    public (Dictionary<string, InternalXrefSpec[]> result, Dictionary<FilePath, InternalXrefSpec[]> fileXrefSpecMap) Build()
     {
+        var fileXrefSpecMap = new ConcurrentDictionary<FilePath, InternalXrefSpec[]>();
         var builder = new ListBuilder<InternalXrefSpec>();
```

Comment:
```suggestion
        var builder = new ListBuilder<(FilePath, InternalXrefSpec)>();
```

It seems better to reuse this builder to keep the immutability of `Load`.

Reply:
Sorry, I did not get your point.

fileXrefSpecMap  is defined as ConcurrentDictionary here but not ListBuilder as the use cases are different, i.e. there are possible uid-conflict issue but no file conflict issue.
https://github.com/freewheel70/docfx/blob/82dd5ea44c39770d12c431bcdfce1215e46399f0/src/docfx/build/xref/InternalXrefMapBuilder.cs#L90